lib/linter/linter.js

```diff
@@ -1770,12 +1775,24 @@ class Linter {
         debug("With flat config: %s", options.filename);
 
         // we need a filename to match configs against
-        const filename = options.filename || "<input>";
+        const filename = options.filename || "__placeholder__.js";
 
         // Store the config array in order to get plugin envs and rules later.
         internalSlotsMap.get(this).lastConfigArray = configArray;
         const config = configArray.getConfig(filename);
 
+        if (!config) {
+            return [
+                {
+                    ruleId: null,
+                    severity: 1,
+                    message: `No matching configuration found for ${filename}.`,
+                    line: 0,
+                    column: 0
+                }
+            ];
+        }
+
```

Comment:
Unlike with eslintrc, it's possible that the filename given might not match something in the config array. If that's the case, I've opted to create a warning saying that the file didn't have a configuration.

I wasn't sure if this should be considered an error, so I used the model of creating a warning to match what we do for ignored files in `ESLint`, but definitely open to other ideas.